lua/toggleterm/ui.lua

```diff
@@ -183,13 +183,27 @@ local split_commands = {
   },
 }
 
+--- @param size number/function
+local function resolve_size(size)
+  if type(size) == 'number' then
+    return function()
+      return size
+    end
+  end
+  if type(size) == 'function' then
+    return size
+  end
+  error(string.format('The input %s is not of type "number" or "function".',size))
+end
+
 --- @param size number
 --- @param term Terminal
 function M.open_split(size, term)
   local has_open, win_ids = M.find_open_windows()
   local commands = split_commands[term.direction]
 
   size = M.get_size(size)
+  size = resolve_size(size)(term)
```

Comment:
This is largely stylistic but I personally find the double invocation less readable than declaring another variable, granted it's slightly more wasteful but I think the syntax is less approachable.

Reply:
I think if you refactor this function to take two arguments then you could just wrap the result of `M.get_size` e.g.
```lua
size = resolve_size(M.get_size(size), term))
```